src/SFA.DAS.Commitments.Application/Commands/CreateApprenticeshipUpdate/CreateApprenticeshipUpdateCommandHandler.cs

```diff
@@ -109,17 +110,21 @@ protected override async Task HandleCore(CreateApprenticeshipUpdateCommand comma
                 await SendApprenticeshipUpdateCreatedEvent(apprenticeship);
             }
 
-            await Task.WhenAll(
+            var tasksToRun = new List<Task>
+            {
                 _apprenticeshipUpdateRepository.CreateApprenticeshipUpdate(pendingUpdate, immediateUpdate),
-                SaveHistory(),
-                _v2EventsPublisher.PublishApprenticeshipUlnUpdatedEvent(immediateUpdate)
-            );
+                SaveHistory()
+            };
 
-            if (command.ApprenticeshipUpdate.ULN != null)
+            if (!string.IsNullOrEmpty(command.ApprenticeshipUpdate.ULN))
```

Comment:
If possible can we please also check that it is the provider who has made the change as employer aren't allowed to make ULN change